utilities/blob_db/blob_db_impl.cc

```diff
@@ -0,0 +1,2112 @@
+          "EVICT COMPACTION FAILURE SN: %d FN: %d OFFSET: %d SIZE: %d",
+          packet.dsn_, packet.file_number_, packet.blob_offset_,
+          packet.blob_size_);
+
+    if (debug_level_ >= 3)
+      Log(InfoLogLevel::INFO_LEVEL, db_options_.info_log,
+          "EVICT COMPACTED SN: %d FN: %d OFFSET: %d SIZE: %d SUCC: %d",
+          packet.dsn_, packet.file_number_, packet.blob_offset_,
+          packet.blob_size_, succ);
+  }
+  return std::make_pair(true, -1);
+}
+
+std::pair<bool, int64_t> BlobDBImpl::evictDeletions(bool aborted) {
+  if (aborted) return std::make_pair(false, -1);
+
+  ColumnFamilyHandle* last_cfh = nullptr;
+  Options last_op;
+
+  Arena arena;
+  ScopedArenaIterator iter;
+  RangeDelAggregator range_del_agg(
+      InternalKeyComparator(ioptions_.user_comparator), {} /* snapshots */);
+
+  delete_packet_t dpacket;
+  while (delete_keys_q_.dequeue(&dpacket)) {
+    if (last_cfh != dpacket.cfh_) {
+      // this can be expensive
+      last_cfh = dpacket.cfh_;
+      last_op = db_impl_->GetOptions(last_cfh);
+      iter.set(
+          db_impl_->NewInternalIterator(&arena, &range_del_agg, dpacket.cfh_));
+      // this will not work for multiple CF's.
+    }
+
+    Slice user_key(dpacket.key_);
+    InternalKey target(user_key, dpacket.dsn_, kTypeValue);
+
+    Slice eslice = target.Encode();
+    iter->Seek(eslice);
+
+    if (!iter->status().ok()) {
+      Log(InfoLogLevel::INFO_LEVEL, db_options_.info_log,
+          "Invalid iterator seek %s", dpacket.key_.c_str());
+      continue;
+    }
+
+    const Comparator* bwc = BytewiseComparator();
+    while (iter->Valid()) {
+      if (!bwc->Equal(ExtractUserKey(iter->key()), ExtractUserKey(eslice)))
+        break;
+
+      ParsedInternalKey ikey(Slice(), 0, kTypeValue);
+      if (!ParseInternalKey(iter->key(), &ikey)) {
+        continue;
+      }
+
+      // once you hit a DELETE, assume the keys below have been
+      // processed previously
+      if (ikey.type == kTypeDeletion || ikey.type == kTypeSingleDeletion) break;
+
+      Slice val = iter->value();
+      markBlobDeleted(ikey.user_key, val);
+
+      iter->Next();
+    }
+  }
+  return std::make_pair(true, -1);
+}
+
+std::pair<bool, int64_t> BlobDBImpl::checkSeqFiles(bool aborted) {
+  if (aborted) return std::make_pair(false, -1);
+
+  std::vector<std::shared_ptr<BlobFile>> process_files;
+  {
+    ReadLock l(&mutex_);
+    for (auto bfile : open_blob_files_) {
+      {
+        ReadLock rl_f(&bfile->mutex_);
+
+        std::time_t epoch_now = std::chrono::system_clock::to_time_t(
+            std::chrono::system_clock::now());
+        if (bfile->ttl_range_.second > epoch_now) continue;
+        process_files.push_back(bfile);
+      }
+    }
+  }
+
+  for (auto bfile : process_files) closeSeqWrite(bfile, false);
+
+  return std::make_pair(true, -1);
+}
+
+std::pair<bool, int64_t> BlobDBImpl::fsyncFiles(bool aborted) {
+  if (aborted) return std::make_pair(false, -1);
```

Comment:
RocksDB doesn't provide users an option to periodically sync WAL. I don't think there is a need for blob storage to do that either. Why do we need to do this?